ibm_was/datadog_checks/ibm_was/ibm_was.py

```diff
@@ -27,45 +27,47 @@ def __init__(self, name, init_config, instances):
             'RangeStatistic': self.gauge,
             'TimeStatistic': self.gauge,
         }
+        self.url = self.instance.get('servlet_url')
+        self.custom_queries = self.instance.get('custom_queries', [])
+        self.custom_queries_units_gauge = set(self.instance.get('custom_queries_units_gauge', []))
+        self.custom_tags = self.instance.get('tags', [])
+        self.collect_stats = self.setup_configured_stats()
+        self.nested_tags, self.metric_categories = self.append_custom_queries()
+        self.custom_stats = set(self.nested_tags)
+        self.service_check_tags = self.custom_tags + ['url:{}'.format(self.url)]
 
-    def check(self, instance):
-        validation.validate_config(instance)
-        collect_stats = self.setup_configured_stats(instance)
-        url = instance.get('servlet_url')
-        self.custom_queries_units_gauge = set(instance.get('custom_queries_units_gauge', []))
+        self.check_initializations.append(self._validate_config)
```

Comment:
(Nit) Should we run this validation in `__init__`, rather than a per-check initialization? Or is there a specific benefit to using check initializations for config validation?

```python
url = instance.get('servlet_url')
if not url:
    raise ConfigurationError(...)

self.url = url
...
```

Reply:
Check initializations raise the exception on every check run so its more prominent on logs. I originally also had an explicit call on init